src/main/java/com/minecolonies/core/datalistener/DiseasesListener.java

```diff
@@ -0,0 +1,234 @@
+            byteBuf.writeInt(disease.rarity);
+            for (final ItemStorage cureItem : disease.cureItems)
+            {
+                StandardFactoryController.getInstance().serialize(byteBuf, cureItem);
+            }
+        }
+        Network.getNetwork().sendToPlayer(new GlobalDiseaseSyncMessage(byteBuf), player);
+    }
+
+    /**
+     * Read the data from the packet and parse it.
+     *
+     * @param byteBuf pck.
+     */
+    public static void readGlobalDiseasesPackets(final FriendlyByteBuf byteBuf)
+    {
+        final RandomCollection<ResourceLocation, Disease> newDiseases = new RandomCollection<>();
+        final int size = byteBuf.readInt();
+        for (int i = 0; i < size; i++)
+        {
+            final ResourceLocation id = byteBuf.readResourceLocation();
+            final Component name = byteBuf.readComponent();
+            final int rarity = byteBuf.readInt();
+
+            final List<ItemStorage> cureItems = new ArrayList<>();
+            final int itemCount = byteBuf.readInt();
+            for (int j = 0; j < itemCount; j++)
+            {
+                cureItems.add(StandardFactoryController.getInstance().deserialize(byteBuf));
+            }
+
+            newDiseases.add(rarity, id, new Disease(id, name, rarity, cureItems));
+        }
+        DISEASES = newDiseases;
+    }
+
+    /**
+     * Get a collection of all possible diseases.
+     *
+     * @return the collection of diseases.
+     */
+    @NotNull
+    public static Collection<Disease> getDiseases()
+    {
+        return DISEASES.getAll();
+    }
+
+    /**
+     * Get a specific disease by id.
+     *
+     * @param id the disease id.
+     * @return the disease instance or null if it does not exist.
+     */
+    @Nullable
+    public static Disease getDisease(final ResourceLocation id)
+    {
+        return DISEASES.get(id);
+    }
+
+    /**
+     * Get a random disease from the list of diseases.
+     *
+     * @param random the random provider.
+     * @return the random disease instance or null if no diseases exist.
+     */
+    @Nullable
+    public static Disease getRandomDisease(final RandomSource random)
+    {
+        return DISEASES.next(random);
+    }
+
+    /**
+     * Predicate for the different usages to check if inventory contains a cure.
+     *
+     * @param cure the expected cure item.
+     * @return the predicate for checking if the cure exists.
+     */
+    public static Predicate<ItemStack> hasCureItem(final ItemStorage cure)
+    {
+        return stack -> isCureItem(stack, cure);
+    }
+
+    /**
+     * Check if the given item is a cure item.
+     *
+     * @param stack the input stack.
+     * @param cure  the cure item.
+     * @return true if so.
+     */
+    public static boolean isCureItem(final ItemStack stack, final ItemStorage cure)
+    {
+        return ItemStackUtils.compareItemStacksIgnoreStackSize(stack, cure.getItemStack(), !cure.ignoreDamageValue(), !cure.ignoreNBT());
```

Comment:
Why compare storage and stack? Then just compare storage and storage and adjust this to get new Itemstorage(stack)